Add Angular elements and web component build

amberramesh opened the pull request:
- Add scripts for packing/concatenating web component
- Add module for creating wc
- Add service to append wc to app root

amberramesh committed:
Add Angular elements and web component build
- Add scripts for packing/concatenating web component
- Add module for creating wc
- Add service to append wc to app root

bherr2 commented on src/app/app.component.html:
```diff
@@ -1,3 +1 @@
-<div id="app">
-  <tissue-blocks></tissue-blocks>
-</div>
+<div id="app"></div>
```
Why can't you use the cns tissue block web component here? Also you don't need id="app" (bad practice in Angular)

amberramesh replied:
Updated to using web component directly (along with related changes).

bherr2 commented on src/app/app.component.ts:
```diff
@@ -1,15 +1,17 @@
-  public constructor(private titleService: Title) {
-    this.titleService.setTitle('Tissue Bar Graphs')
+  public constructor(private titleService: Title, private tissueBlockService: TissueBlocksService) {
+    this.titleService.setTitle('CNS Tissue Blocks')
```
Remove this and just set the title in the index.html

bherr2 commented on src/app/app.component.ts:
```diff
@@ -1,15 +1,17 @@
 })
 export class AppComponent {
-  title = 'tissue-bar-graphs'
+  title = 'tissue-blocks'
```
Remove this and just set the title in the index.html

bherr2 commented on src/app/app.component.ts:
```diff
@@ -1,15 +1,17 @@
-    this.titleService.setTitle('Tissue Bar Graphs')
+  public constructor(private titleService: Title, private tissueBlockService: TissueBlocksService) {
+    this.titleService.setTitle('CNS Tissue Blocks')
+    this.tissueBlockService.injectElement()
```
You shouldn't need to do this. You should be able to just use the custom element in your app.component.html

amberramesh replied:
Have set the document title instead of using service.

bherr2 commented on src/app/services/tissue-blocks.service.ts:
```diff
@@ -0,0 +1,11 @@
+import { TissueBlocksComponent } from '../components/tissue-blocks.component'
+
+@Injectable()
+export class TissueBlocksService {
```
Don't do this.

bherr2 reviewed (changes requested):
Also, you need to update/add instructions for embedding the component to the README.md

amberramesh committed:
Remove service for appending wc
- Include wc in app template
- Set title for app in document instead of using Title service

amberramesh committed:
Reconfigure initialization and data loading
- Configuration can be passed to tissue blocks component/wc
- Additional option to show configuration UI
- Move data sources to Google sheets from local directories
- Add tsconfig for wc, set styles for host component

amberramesh committed:
Update README with web component info

amberramesh committed:
Add build workflow for GitHub Actions

amberramesh committed:
Add usage example

amberramesh committed:
Fix wc build description
